src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -158,6 +167,15 @@ export class IndexPattern implements IIndexPattern {
     }
   }
 
+  private populateDataSourceReference(dataSourceRef?: SavedObjectReference) {
+    return dataSourceRef
+      ? {
+          ...dataSourceRef,
+          name: DATA_SOURCE_REFERNECE_NAME,
```

Comment:
this name here, will overwrite the name property in the original dataSourceRef obj. Is that what you want?

Reply:
yes